### PR TITLE
Warn on MaximumReceiveMessageSize

### LINKED ISSUE #27013: Warn on MaximumReceiveMessageSize size (DOS risk)
Per https://github.com/dotnet/aspnetcore/issues/43974#issuecomment-1247486989.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0969556b-b77e-bdb8-8e20-fd6971be0c25
* Version Independent ID: d17e4c28-ea82-0dd8-295d-19f6ada147ea
* Content: [ASP.NET Core Blazor SignalR guidance](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/signalr?view=aspnetcore-6.0)
* Content Source: [aspnetcore/blazor/fundamentals/signalr.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/fundamentals/signalr.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #27013

### CODE REVIEW

**Comment by guardrex** [comment]
Let's hold off on review. We're discussing this on the PU issue. We might do something different ... an INCLUDE used in the Blazor doc and the SignalR doc, for example. We might do this, but we might only do it in the SignalR doc.

**Comment by guardrex** [comment]
Ok ... well ... the PU issue was closed right in the middle of the discussion on this.

Let's go with *Rick's Gambit*&trade; ... i.e. ... **_Call for review!_** 😆

Brennan, which way would you like to go?

* Just note this on the Blazor-side, as this PR does.
* Create a duplicate NOTE for the SignalR config topic ... I'll add it to this PR.
* Do something else.

**Comment by BrennanConroy** [comment]
> * I can place it in the **Description** in the table at ...

Sure, similar to how `ApplicationMaxBufferSize` in a later table mentions memory usage can increase if the setting is changed.

**Comment by guardrex** [comment]
@BrennanConroy ... Added a remark to the table. How's it look?

File: aspnetcore/blazor/fundamentals/signalr.md
Comment (TanayParikh) [inline aspnetcore/blazor/fundamentals/signalr.md:170]: I believe 32KB should be sufficient unless we follow the convention of writing total bytes elsewhere?

```suggestion
> We recommend setting <xref:Microsoft.AspNetCore.SignalR.HubOptions.MaximumReceiveMessageSize> to a small value in order to limit the risk of [Denial of service (DoS) attacks](xref:blazor/security/server/threat-mitigation#denial-of-service-dos-attacks). The default value is 32 KB.
```

File: aspnetcore/blazor/security/server/threat-mitigation.md
Comment (TanayParikh) [inline aspnetcore/blazor/security/server/threat-mitigation.md:95]:
```suggestion
[Denial of service (DoS) attacks](https://developer.mozilla.org/docs/Glossary/DOS_attack) involve a client causing the server to exhaust one or more of its resources making the app unavailable. Blazor Server apps include default limits and rely on other ASP.NET Core and SignalR limits that are set on <xref:Microsoft.AspNetCore.Components.Server.CircuitOptions> to protect against DoS attacks:
```

**Review by TanayParikh — COMMENTED** [review]
LGTM, will leave the final sign-off for @BrennanConroy

File: aspnetcore/blazor/fundamentals/signalr.md
Comment (guardrex) [reply aspnetcore/blazor/fundamentals/signalr.md:170]: > the convention

That's the problem. The standard was never adopted. They wrote a whole page on how to use bit/byte terms without even addressing it ...

https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/term-collections/bits-bytes-terms

I think ur aware technically (IEC and SI) that we should be using "KiB" (kibibyte) to eliminate the confusion (or "kB" for multiples of 1,000). Since they won't permit that approach, this is how I've been hardening docs to avoid confusion.

Comment (guardrex) [reply aspnetcore/blazor/fundamentals/signalr.md:170]: I might change it tho ... I'm going to open an issue to discuss this with the docs 🐈 🐈 🐈.

I performed a search to see how consistent I've been. It looks like in more than half the cases, I did write out either format (X bytes or X * 1024). It looks like a number of cases without it exist, mostly in content sent over from the PU that I didn't change on publication.

I'll circle around and update depending on the outcome of that discussion ... but let's not hold this up over that.

**Review by TanayParikh — APPROVED** [review]
LGTM, but I'd await @BrennanConroy's signoff.

**Comment by guardrex** [comment]
I will ... and btw ... I've talked myself out of using explicit byte conventions, as you recommended, on the issue that I opened ...

https://github.com/dotnet/AspNetCore.Docs/issues/27056

I wish MS/.NET foundation would adopt the standard. What I can do is place a remark on it in the Blazor *Fundamentals* topic and just use "KB"/"MB"/"GB" everywhere.

**Comment by guardrex** [comment]
@BrennanConroy ... Are you OOF? No rush ... just checking.

File: aspnetcore/signalr/configuration.md
Comment (BrennanConroy) [inline aspnetcore/signalr/configuration.md:450]: nit: the wording "We recommend setting" makes it sound like the user /should/ change the value.
"We recommend keeping the value small", "Increasing this value may increase the risk of DOS...", etc.

Comment (guardrex) [reply aspnetcore/signalr/configuration.md:450]: The 2nd one sounds good :ear: to me. I'll try that on the next commit.

**Comment by guardrex** [comment]
@BrennanConroy ... Ok ... ready for another 👁️.

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**
